apps/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -0,0 +1,588 @@
+            )
+          ) {
+            console.log();
+            console.log(
+              colors.red(
+                `"${rushProject.packageName}" depends on package "${name}" (${version}) which exists ` +
+                  'within the workspace but cannot be fulfilled with the specified version range. Either ' +
+                  'specify a valid version range, or add the package as a cyclic dependency.'
+              )
+            );
+            throw new AlreadyReportedError();
+          }
+
+          if (!this.options.allowShrinkwrapUpdates) {
+            console.log();
+            console.log(
+              colors.red(
+                `"${rushProject.packageName}" depends on package "${name}" (${version}) which exists within ` +
+                  'the workspace. Run "rush update" to update workspace references for this package.'
+              )
+            );
+            throw new AlreadyReportedError();
+          }
+
+          // We will update to `workspace` notation. If the version specified is a range, then use the provided range.
+          // Otherwise, use `workspace:*` to ensure we're always using the workspace package.
+          const workspaceRange: string =
+            !!semver.validRange(dependencySpecifier.versionSpecifier) &&
+            !semver.valid(dependencySpecifier.versionSpecifier)
+              ? dependencySpecifier.versionSpecifier
+              : '*';
+          packageJson.addOrUpdateDependency(name, `workspace:${workspaceRange}`, dependencyType);
+          shrinkwrapIsUpToDate = false;
+          continue;
+        } else if (dependencySpecifier.specifierType === 'workspace') {
+          // Already specified as a local project. Allow the package manager to validate this
+          continue;
+        }
+
+        // Allow the package manager to handle peer dependency resolution, since this is simply a constraint
+        // enforced by the package manager
+        if (dependencyType === DependencyType.Peer) {
+          continue;
+        }
+
+        // It is not a local dependency, validate that it is compatible
+        if (
+          shrinkwrapFile &&
+          !shrinkwrapFile.hasCompatibleWorkspaceDependency(
+            dependencySpecifier,
+            shrinkwrapFile.getWorkspaceKeyByPath(
+              this.rushConfiguration.commonTempFolder,
+              rushProject.projectFolder
+            )
+          )
+        ) {
+          shrinkwrapWarnings.push(
+            `Missing dependency "${name}" (${version}) required by "${rushProject.packageName}"`
+          );
+          shrinkwrapIsUpToDate = false;
+        }
+      }
+
+      // Save the package.json if we modified the version references and warn that the package.json was modified
+      if (packageJson.saveIfModified()) {
+        console.log(
+          colors.yellow(
+            `"${rushProject.packageName}" depends on one or more workspace packages which did not use "workspace:" ` +
```

Comment:
Okay there's a usability problem here:
1. Set `"useWorkspaces": true`
2. Run `rush update` and get these warnings
3. Set `"useWorkspaces": false`
4. Run `rush update` again

- No warning is printed, and the `workspace:` prefix is not automatically removed.
- As a result, the install fails with a cryptic error: `ERROR: Cannot get dependency key for temp project: @rush-temp/my-app`

Reply:
This would be easy enough to add